src/engine.h

```diff
@@ -104,11 +132,20 @@ class Engine {
   // This covers both Vulkan buffers and images.
   virtual Result DoBuffer(const BufferCommand* cmd) = 0;
 
-  // Run all queued commands and copy result data to the host.
-  virtual Result DoProcessCommands(uint32_t* stride,
-                                   uint32_t* width,
-                                   uint32_t* height,
-                                   const void** buf) = 0;
+  // Run all queued commands and copy frame buffer data to the host
+  // if graphics pipeline.
+  virtual Result DoProcessCommands() = 0;
+
+  // Get stride, width, height, and memory pointer of frame buffer.
+  virtual Result GetFrameBufferInfo(ResourceInfo* info) = 0;
```

Comment:
Should say this is only valid if the framebuffer buffer is mapped into the host address space. In particular, if we have run DoProcessCommands() and that since then no graphics pipeline drawing commands have occurred (such as DoClear, DoDrawArrays, DoDrawRect).